testing/src/main/java/org/oppia/android/testing/logging/EventLogSubject.kt

```diff
@@ -1944,6 +1964,56 @@ class EventLogSubject private constructor(
     }
   }
 
+  /**
+   * Truth subject for verifying properties of [EventLog.FeatureFlagContext]s.
+   *
+   * Note that this class is also a [LiteProtoSubject] so other aspects of the underlying
+   * [EventLog.FeatureFlagContext] proto can be verified through inherited methods.
+   *
+   * Call [FeatureFlagContextSubject.assertThat] to create the subject.
+   */
+  class FeatureFlagContextSubject private constructor(
+    metadata: FailureMetadata,
+    private val actual: EventLog.FeatureFlagContext
+  ) : LiteProtoSubject(metadata, actual) {
+    /**
+     * Returns a [StringSubject] to test [EventLog.FeatureFlagContext.getSessionId].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasSessionIdThat(): StringSubject = assertThat(actual.sessionId)
+
+    /**
+     * Returns a [BooleanSubject] to test the flagEnabledState of the first item on the feature flag
+     * list in the [EventLog.FeatureFlagContext].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasFeatureEnabledStateThat(): BooleanSubject =
+      assertThat(actual.getFeatureFlag(0).flagEnabledState)
+
+    /**
+     * Returns a [ComparableSubject] to test the flagSyncStatus of the
+     * [EventLog.FeatureFlagContext].
+     *
+     * This method never fails since the underlying property defaults to empty string if it's not
+     * defined in the context.
+     */
+    fun hasSyncStatusThat(): ComparableSubject<PlatformParameter.SyncStatus> =
+      assertThat(actual.getFeatureFlag(0).flagSyncStatus)
+
+    companion object {
+      /**
+       * Returns a new [SurveyContextSubject] to verify aspects of the specified
+       * [EventLog.SurveyContext] value.
+       */
```

Comment:
Update references to survey

Reply:
I have updated the comments here for better clarity and removed the Survey references.